v2/client/src/components/pages/survey/Survey.js

```diff
@@ -45,10 +39,28 @@ export default class Survey extends React.Component {
         });
       })
       .then(() => {
```

Comment:
this does not have the setstate will finish before the next .then

if you need to guarantee execution order, you need to pass a callback in setstate after the object